tools/build_defs/framework.bzl

```diff
@@ -518,25 +525,15 @@ def _correct_path_variable(env):
     env["PATH"] = "$PATH:" + value
     return env
 
-def _depset(item):
-    if item == None:
-        return depset()
-    return depset([item])
-
-def _list(item):
-    if item:
-        return [item]
-    return []
-
 def _copy_deps_and_tools(files):
     lines = []
     lines += _symlink_contents_to_dir("lib", files.libs)
     lines += _symlink_contents_to_dir("include", files.headers + files.include_dirs)
 
     if files.tools_files:
         lines.append("##mkdirs## $$EXT_BUILD_DEPS$$/bin")
-    for tool in files.tools_files:
-        lines.append("##symlink_to_dir## $$EXT_BUILD_ROOT$$/{} $$EXT_BUILD_DEPS$$/bin/".format(tool))
+    for tool in files.tools_files.to_list():
+        lines.append("##symlink_to_dir## $$EXT_BUILD_ROOT$$/{} $$EXT_BUILD_DEPS$$/".format(_file_path(tool)))
```

Comment:
I actually think that by using `$(execpath)` in places that this code can be removed. I need to add calls to `ctx.expand_location` for the command strings so they can use the tools provided here directly. 
I also think that introducing a global `env` attribute, as we've discussed in other PRS, and then allowing Make variable substitution of these variables into the commands would be a good idea.